lib/src/models/comment_impl.dart

```diff
@@ -150,10 +149,56 @@ class MoreComments extends RedditBase with RedditBaseInitializedMixin {
       _comments = await reddit.post(apiPath['morechildren'], data);
 
       if (update) {
-        _comments.forEach((c) => c._submission = _submission);
+        _comments.forEach((c) {
+          c._submission = _submission;
+        });
       }
     }
-    return _comments;
+    return _fillCommentsForests(_comments);
+  }
+
+  List<dynamic> _fillCommentsForests(List<dynamic> fullList) {
+    final first = fullList.first;
+    if (fullList.length > 0 && first is Comment) {
```

Comment:
This should be:

```dart
if (fullList.isNotEmpty && (fullList.first is Comment)) {
```

Reply:
Why are we explicitly checking that the first element is `Comment`? It's not obvious to me.

Reply:
I agree that I should make it more obvious. This is basically checking for MoreComments, which shouldn’t be resolved here. More Comments also signify the end of the layer.

From my understanding, if the list only contains one Comment or one MoreComment, we don’t have to resolve anything, so maybe I should just change to length > 1?

Reply:
Yeah, I think that makes sense (I haven't looked at the `Comment` code for awhile). Thanks for clearing that up!